### ISSUE TITLE
Missing string escaping for \r

### ISSUE THREAD

**Issue by non** [scalameta/munit#258]
From cats-parse I'm seeing this output:

```
  + wsp rejects all but single char 0.001s
\n 0.0sf accepts
\n 0.003srejects all but
  + lwsp accepts all linear white space 0.011s
```

The original test names are `"crlf accepts \r\n"` and `"crlf rejects all but \r\n"`.

I think munit is not escaping `\r` properly (although it does seem to escape `\n`). The caller can definitely fix this by escaping their own strings but it seems friendly to have munit do this (especially if test names contain unexpected characters).

**Comment by olafurpg** [comment]
Thank you for reporting! This should be fixed in #261, the test report for a suite that stresses escaping now looks like this

<img width="295" alt="Screenshot 2020-11-18 at 06 45 38" src="https://user-images.githubusercontent.com/1408093/99491531-378b1280-296c-11eb-91c0-781319bf6cf6.png">

**Comment by olafurpg** [comment]
Kickstarted https://github.com/scalameta/munit/releases/tag/v0.7.18
